src/bson_value.ts

```diff
@@ -10,8 +11,16 @@ export abstract class BSONValue {
     return BSON_MAJOR_VERSION;
   }
 
+  [Symbol.for('nodejs.util.inspect.custom')](
+    depth?: number,
+    options?: unknown,
+    inspect?: InspectParameterFn
+  ): string {
+    return this.inspect(depth, options, inspect);
```

Comment:
Let's chat on slack about this because I think I may be overlooking some complexity, could this function be responsible for defaulting the inspect function when it is not provided, and could the defaultInspect function include the logic for "addQuotes"?

Reply:
From IRL: We can continue to rely on JSON.stringify as we had prior it will handle the quoting and escaping for us as long as the original type is passed in.

Ideally, we can handle ES Map and bigint in a replacer function without too much complexity